content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-gitlab-identity.md

```diff
@@ -0,0 +1,423 @@
+
+```sh
+terraform plan
+```
+
+Then apply the configuration.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 4 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + gitlab-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `gitlab-ci` value.
+
+```
+. . .
+
+pply complete! Resources: 4 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+gitlab-identity = "<your actions identity>"
+```
+
+This is the identity's [UIDP (unique identity path)](/chainguard/chainguard-enforce/reference/events/#uidp-identifiers), which you configured the `gitlab.tf` file to emit in the previous section. Note this value down, as you'll need it to set up the GitLab CI pipeline you'll use to test the identity. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
+
+```sh
+chainctl iam identities ls
+```
+
+Note that you may receive a `PermissionDenied` error part way through the apply step. If so, run `chainctl auth login` once more, and then `terraform apply` again to resume creating the identity and resources.
+
+You're now ready to create a GitLab CI pipeline which you'll use to test out this identity.
+
+
+## Testing the identity with a GitLab CI/CD Pipeline
+
+From the GitLab Dashboard, select **Projects** in the left-hand sidebar menu. From there, click on the project you specified in the `subject` claim to be taken to the project overview. There, in the list of the repository's contents, there will be a file named `.gitlab-ci.yml`. This is a special file that's required when using GitLab CI/CD, as it contains the CI/CD configuration. 
+
+Click on the `.gitlab-ci.yml` file, then click the **Edit** button and select an option for editing the file. For the purpose of this guide, delete whatever content is in this file to start and replace it with the following. 
+
+```
+image: cgr.dev/chainguard/wolfi-base
+
+stages:
+  - assume-and-explore
+
+assume-and-explore:
+  id_tokens:
+    ID_TOKEN_1:
+      aud: https://gitlab.com
+
+  stage: assume-and-explore
+
+  script:
+  - |
+    # Install chainctl
+    wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"
+    chmod +x chainctl
+    mv chainctl /usr/bin
+
+    # Assume
+    chainctl auth login \
+      --identity-token $ID_TOKEN_1 \
+      --identity <your gitlab identity>
+
+    # Explore
+    chainctl policy ls
+```
+
+Let's go over what this configuration does. 
+
+GitLab requires that pipelines have a shell. To this end, this configuration uses the `wolfi-base` image
```

Comment:
```suggestion
First, GitLab requires that pipelines have a shell. To this end, this configuration uses the [`cgr.dev/chainguard/wolfi-base` image](https://edu.chainguard.dev/chainguard/chainguard-images/reference/wolfi-base/overview/) since it includes the `sh` shell.
```

Reply:
Woof, I just left this sentence unfinished! Thanks for fleshing it out